langs/iogii/iogiiwrapper.c

```diff
@@ -0,0 +1,71 @@
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+#define ERR_AND_EXIT(msg) do { perror(msg); exit(EXIT_FAILURE); } while (0)
+
+const char* iogii = "/usr/bin/iogii", *code = "code.iog", *input = "argv.txt";
+
+int main(int argc, char* argv[]) {
+    if (!strcmp(argv[1], "--version")) {
+        execv(iogii, argv);
+        ERR_AND_EXIT("execv");
+    }
+
+    if (chdir("/tmp"))
+        ERR_AND_EXIT("chdir");
+
+    FILE* fp;
+
+    if (!(fp = fopen(code, "w")))
+        ERR_AND_EXIT("fopen");
+
+    char buffer[4096];
+    ssize_t nbytes;
+
+    while ((nbytes = read(STDIN_FILENO, buffer, sizeof(buffer))))
+        if (fwrite(buffer, sizeof(char), nbytes, fp) != (size_t) nbytes)
+            ERR_AND_EXIT("fwrite");
+
+    if (fclose(fp))
+        ERR_AND_EXIT("fclose");
+
+    int fd;
+
+    if (!(fd = open(input, O_CREAT | O_TRUNC | O_WRONLY, 0644)))
+        ERR_AND_EXIT("open");
+
+    for (int i = 2; i < argc; i++)
+        if (write(fd, argv[i], strlen(argv[i])) < 0 || write(fd, "\n", sizeof(char)) < 0) {
+            if (close(fd))
+                ERR_AND_EXIT("close");
+
+            ERR_AND_EXIT("write");
+        }
+
+    if (close(fd))
+        ERR_AND_EXIT("close");
+
+    pid_t pid;
+
+    if (!(pid = fork())) {
+        if (dup2(open(input, O_RDONLY), STDIN_FILENO))
```

Comment:
iogii can accept args as a [[char]] if you preceed them with --

e.g. iogii -- arg1 arg2 arg3 
now $ is ["arg1","arg2","arg3"]

I think this would be better since args can contain newlines too and that would be ambiguous with different args?

Reply:
Addressed with commit [a0b0d54](https://github.com/code-golf/code-golf/pull/1765/commits/a0b0d540a2dc9a052cea28031f2f2840f9f1a746). This is way better, absolutely, and fully resembles the approach that comes in second place for handling command-line arguments on the repository.

Was this introduced with 0.2 by any chance?

Reply:
Great! And yeah I introduced it in 0.2 specifically for this purpose.

If this is the second place way for handling command line arguments, what is the first place way?

Reply:
`-e`, `--eval`, `--evaluate`, `--exec`, `--execute`, `--run`, ...
```bash
/usr/bin/iogii -e <code> -- <arguments>
```

No project structure, no source file, no code runner, just REPL.